CMakeLists.txt

```diff
@@ -25,13 +25,32 @@ target_include_directories(${PROJECT_NAME}
       ${PROJECT_SOURCE_DIR}/include
 )
 
+target_sources(${PROJECT_NAME}
+    PRIVATE
+        ${PROJECT_SOURCE_DIR}/include/ipr/ancillary
+        ${PROJECT_SOURCE_DIR}/include/ipr/attribute
+        ${PROJECT_SOURCE_DIR}/include/ipr/cxx-form
+        ${PROJECT_SOURCE_DIR}/include/ipr/impl
+        ${PROJECT_SOURCE_DIR}/include/ipr/interface
+        ${PROJECT_SOURCE_DIR}/include/ipr/io
+        ${PROJECT_SOURCE_DIR}/include/ipr/lexer
```

Comment:
`ipr/interface`, `ipr/impl`, and `ipr/io` should be considered `INTERFACE` files.